spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+    private static OwnerProductApi ownerProductApi;
+    private static OwnerContentApi ownerContentApi;
+    private static ConsumerClient consumerApi;
+    private static final String STACKED_VIRT_POOL_1 = "123";
+    private static final String STACKED_VIRT_POOL_2 = "456";
+    private static final String STACKED_VIRT_POOL_3 = "444";
+    private static final String STACKED_NON_VIRT_POOL = "789";
+    private static final String NON_STACKED_POOL = "234";
+    private static final String DATACENTER_POOL = "222";
+    private static final String REGULAR_STACKED_WITH_DIFF_STACK_ID = "888";
+
+    @BeforeAll
+    public static void beforeAll() {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+        ownerProductApi = client.ownerProducts();
+        ownerContentApi = client.ownerContent();
+        consumerApi = client.consumers();
+    }
+
+    @Test
+    public void shouldCreateTheProperPools() {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+
+        String stackId1 = StringUtil.random("stack");
+        String stackId2 = StringUtil.random("stack");
+
+        createVirtLimitProductPools(owner, stackId1);
+        createVirtLimitProduct2Pool(owner, stackId1);
+        createRegularStackedProductPool(owner, stackId1);
+        createNonStackedProductPool(owner);
+        createStackedDatacenterProductPool(owner, stackId1);
+        createStackedProductDiffIdPool(owner, stackId2);
+
+        List<PoolDTO> createdPools = userClient.owners().listOwnerPools(
+            owner.getKey()).stream()
+            .filter(x -> !x.getType().equals("UNMAPPED_GUEST"))
+            .collect(Collectors.toList());
+        assertThat(createdPools).hasSize(7);
+
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(STACKED_VIRT_POOL_1))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(STACKED_VIRT_POOL_2))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(STACKED_VIRT_POOL_3))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(STACKED_NON_VIRT_POOL))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(NON_STACKED_POOL))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream().filter(x -> x.getContractNumber().equals(DATACENTER_POOL))
+            .collect(Collectors.toList())).hasSize(1);
+        assertThat(createdPools.stream()
+            .filter(x -> x.getContractNumber().equals(REGULAR_STACKED_WITH_DIFF_STACK_ID))
+            .collect(Collectors.toList())).hasSize(1);
```

Comment:
You can flip all of these equal checks around to avoid a NPE.
Example: 
```
assertThat(createdPools.stream().filter(x -> STACKED_VIRT_POOL_1.equals(x.getContractNumber()))
            .collect(Collectors.toList())).hasSize(1);
```